### PR TITLE
Shunted Experiments

### PR DESCRIPTION
**Pull request by drkostas**

### CODE REVIEW

File: viz.py [hunk 1]
```diff
@@ -104,8 +104,14 @@ def prepare_model(
...
     args = vars(checkpoint["args"])
+    if 'print_level' in args:
+        args['print_level'] = 3
     print("args:", args)
-    model = getattr(models_mae, arch)(**args)
+    try:
+        model = getattr(models_mae, arch)(**args)
+    except AssertionError as e:
+        print("Error: ", e)
+        return None
```
Comment (andreicozma1) [inline viz.py:114]: Good catch, I needed this done eventually

File: viz.py [hunk 2]
```diff
@@ -167,14 +176,23 @@ def run_one_image(img, model, seed: Optional[int] = None):
     x = torch.einsum("nhwc->nchw", x)
 
     # run MAE
-    _, y, mask = model(x.float(), mask_ratio=0.75, mask_seed=seed)
+    if 'mask_ratio' in model.__dict__:
+        mask_ratio = model.mask_ratio
+    else:
+        mask_ratio = 0.75
+    _, y, mask = model(x.float(), mask_ratio=mask_ratio, mask_seed=seed)
```
Comment (andreicozma1) [inline viz.py:184]: also good catch, mask ratio should be consistent with that defined in args.
May I suggest - should we log "WARN: mask_ratio not found in model config. Defaulting to {mask_ratio}." in the fallback (else) case?

File: train_shunted.sh [hunk 3]
```diff
@@ -23,8 +23,8 @@ ATTENTION="shunted"
 # MODEL_NAME="shunted_2s_mae_vit_mini"
 # MODEL_NAME="shunted_2s_mae_vit_small"
 # MODEL_NAME="shunted_2s_mae_vit_base"
-# MODEL_NAME="shunted_2s_mae_vit_tiny_cross"
-MODEL_NAME="shunted_2s_mae_vit_small_cross"
+MODEL_NAME="shunted_2s_mae_vit_tiny_cross"
```
Comment (andreicozma1) [inline train_shunted.sh:26]: not super important, perhaps starting all variants of the models with the same prefix would help later on with our charts & graphs
Like:
- mae_vit_small
- mae_vit_small_cross
- mae_vit_small_shunted_cross
- etc...

What do you think? 

If you think this is worth pursuing, for backwards compatibility we could do:
image

Comment (drkostas) [reply train_shunted.sh:26]: Good point, I was also thinking of changing that eventually, I'll fix it

File: viz.py (hunk 1 above)
Comment (drkostas) [reply viz.py:114]: Almost forgot to mention it, I had to also add this in the notebook:
```python
# if value is None remove it:
for key in models:
    if models[key] is None:
        del models[key]
```

**Review by andreicozma1 — APPROVED** [review]

**Review by drkostas — COMMENTED** [review]
I think I implemented all your suggestions, I'm merging it

**Pull request merged**